community_build.sh

```diff
@@ -0,0 +1,146 @@
+            "dotty-cps-async" 
+            "scalaz" 
+            "endpoints4s" 
+            "cats-effect-3" 
+            "scala-parallel-collections" 
+            "scala-collection-compat"
+            "scala-java8-compat" 
+            "verify"  
+            "discipline"  
+            "discipline-munit" 
+            "discipline-specs2" 
+            "simulacrum-scalafix"  
+            "cats" 
+            "cats-mtl" 
+            "coop" 
+            "Equal" 
+            "FingerTree"
+            "Log" 
+            "Model" 
+            "Numbers" 
+            "Serial" 
+            "AsyncFile" 
+            "Span" 
+            "scala-stm" 
+            "Lucre" 
+            "izumi-reflect" 
+            "perspective" 
+            "akka" 
+            "Monocle" 
+            "protoquill" 
+            "onnx-scala" 
+            "play-json"
+            "munit-cats-effect" 
+            "scalacheck-effect" 
+            "fs2"  
+            "libretto"  
+            "jackson-module-scala"  
+            "specs2" 
+            "spire" 
+            "http4s" 
+            "parboiled2"
+        )
+
+#set the current folder name
+current_dir=test_community_build
+
+#set output to be the name of the directory that project.log and result_summary.log will be stored
+output=output_no_change
+
+#remove the output directory if it already exits and make a new one
+rm -r "${output}"
+mkdir "${output}"
+
+#set error to be the name of the directory that project.error and error_summary will be stored
+error=error_no_change
+
+#remove the error  directory if it already exits and make a new one
+rm -r "${error}"
+mkdir "${error}"
+
+#create two array for storing the name of project that fail and success
+project_success=()
+project_fail=()
+
+#print the header of error_summary
+printf "%-30s %-10s\n" "Project Name" "# or Errors (explicit_nulls)" >> "${error}/error_summary_count"
+
+summarize_error() {
+     # write all the error messages to project.error
+    grep "\[error\] .*" "${current_dir}/${output}/$1.log" >> "${current_dir}/${error}/error_$1.log"
+    echo -e "\n !!! Error found for project $1: " >> "${current_dir}/${error}/error_summary_message.log"
+    grep "\[error\] .*" "${current_dir}/${output}/$1.log" >> "${current_dir}/${error}/error_summary_message.log"
+
+    # write the error # to error_summary
+    error_line=$(grep -E -m 1 "\[error\] [[:alpha:]]+ errors found" "${current_dir}/${output}/$1.log")
+    if [ -z "${error_line}"]; then
+        error_line=$(grep -E -m 1 "\[error\] [[:alpha:]]+ error found" "${current_dir}/${output}/$1.log")
+    fi
+    error_count=$(echo "$error_line" | cut -d' ' -f2 | grep .)
+
+    printf "%-30s %-10s\n" "$1" "${error_count}" >> "${current_dir}/${error}/error_summary_count"
+}
+
+cd ..
+#run each project in the project_list; add the project to one of project_success or project_fail base on the exit_status
+for project in "${project_list[@]}" 
+do
+    #build the community project
```

Comment:
Just to allow the user to better know where in the list they are at, I'd recommend echoing out something like "Current compiling <whatever>...". if not the script just sits and looks like it's hanging since all the output is being redirected.